src/plugins/spec/spec-new.c

```diff
@@ -0,0 +1,754 @@
+	return matches;
+}
+#endif
+
+/**
+ * Replace the {@link searchFor} with {@link c} in the {@link str}.
+ * The {@link newStr} contains the string with the replaced character.
+ *
+ * @param str the old string
+ * @param newStr the newly created string with the already replaced character
+ * @param searchFor the character to replace
+ * @param c the character to use instead
+ */
+static void replaceCharacter (const char * str, char * newStr, const char searchFor, const char c)
+{
+	for (size_t i = 0; i < elektraStrLen (str); i++)
+	{
+		if (str[i] == searchFor)
+		{
+			newStr[i] = c;
+		}
+		else
+		{
+			newStr[i] = str[i];
+		}
+	}
+}
+
+/**
+ * Appends a key to the KeySet with the default value specified in the meta key (meta:/default) in the
+ * {@link specKey}.
+ *
+ * The default key is added to the `default:/` namespace.
+ *
+ * @param ks the key store to append the new default key to
+ * @param parentKey of the key to appended
+ * @param specKey specification key with meta data of the new default key
+ */
+static void addDefaultKey (KeySet * ks, Key * parentKey, Key * specKey)
+{
+	const Key * defaultMetaKey = keyGetMeta (specKey, "default");
+	const char * defaultValue = keyString (defaultMetaKey);
+
+	const char * parentKeyName = strchr (keyName (parentKey), '/');
+	const char * specKeyName = keyBaseName (specKey);
+
+	char * formattedKeyName = elektraFormat ("default:/%s/%s", parentKeyName, specKeyName);
+	Key * newDefaultKey = keyNew (formattedKeyName, KEY_VALUE, defaultValue, KEY_END);
+	keyCopyAllMeta (newDefaultKey, specKey);
+
+	ksAppendKey (ks, newDefaultKey);
+
+	elektraFree (formattedKeyName);
+}
+
+/**
+ * Check if the specification key has a meta key `required`.
+ *
+ * @param specKey the specification key to check for the required meta key
+ * @return true - if the specification key contains a meta key required
+ * 	   false - if the specification key does not contain a meta key required
+ */
+static bool isRequired (Key * specKey)
+{
+	const Key * key = keyGetMeta (specKey, "require");
+
+	if (key == 0)
+	{
+		return false;
+	}
+
+	const char * keyValue = keyString (key);
+
+	return elektraStrCmp (keyValue, "true") == 0;
+}
+
+/**
+ * Check if the specification key has a meta key `default`.
+ *
+ * @param specKey the specification key to check for the default meta key
+ * @return true - if the specification key contains a meta key default
+ * 	   false - if the specification key does not contain the meta key default
+ */
+static bool hasDefault (Key * specKey)
+{
+	return keyGetMeta (specKey, "default") != 0;
+}
+
+/**
+ * Extracts all the specification keys from the KeySet. After applying this method the {@link ks}
+ * contains no specification namespace.
+ *
+ * It returns the extracted specification keys.
+ *
+ * @param ks the KeySet to extract specification keys from
+ *
+ * @return the extracted specification keys
+ */
+static KeySet * extractSpecKeys (KeySet * ks)
+{
+	KeySet * specKeys = ksNew (0, KS_END);
```

Comment:
```suggestion
	Key * specKey = keyNew ("spec:/", KEY_END);
	KeySet * specKeys = ksCut (ks, specKey);
```

Would be the entire function, no?

Reply:
Yes, you are right!